src/ui/public/courier/data_source/_abstract.js

```diff
@@ -389,6 +391,8 @@ export default function SourceAbstractFactory(Private, Promise, PromiseEmitter)
             recurse(agg.aggs || agg.aggregations);
           });
         }(flatState.body.aggs || flatState.body.aggregations));
+
+        highlightBody(flatState.body);
```

Comment:
I also just realized this would overwrite any highlighting defined via the SearchSource.